apis/task.py

```diff
@@ -256,15 +254,133 @@ def post_process(self) -> DAGNode:
+    """Provision a GPU accelerator via a resource creation.
+
+    Generates a random GPU name and SSH keys, creates a VM Resource, and
+    runs the test config's setup script on the GPU when it is ready.
+
+    Returns:
+      A DAG node that will provision a GPU, an XCome value of the ip address
+      for the host, an XCom value for the GPU name, and an XCom value for
+      the SSH keys.
+
+    Raises:
+      AirflowTaskTimeout: An error occurs when execution_timeout is breached.
+    """
+    with TaskGroup(group_id="provision") as group:
+      with TaskGroup(group_id="initialize"):
+        gpu_name = gpu.generate_gpu_name()
+        ssh_keys = ssh.generate_ssh_keys()
+
+      ip_address = gpu.create_resource.override(task_id="create_resource")(
+          gpu_name,
+          self.image_project,
+          self.image_family,
+          self.task_test_config.accelerator,
+          self.task_gcp_config,
+          ssh_keys,
+      )
+
+      gpu.ssh_host.override(task_id="setup")(
+          ip_address,
+          self.task_test_config.setup_script,
+          ssh_keys,
+      )
+    return group, ip_address, gpu_name, ssh_keys
+
+  def run_model(
+      self,
+      resource: airflow.XComArg,
+      ssh_keys: airflow.XComArg,
+  ) -> DAGNode:
+    """Run the GPU test in `task_test_config`.
+
+    Args:
+      gpu_name: XCom value for the GPU name (string).
+      ssh_keys: And XCom value for the GPU's SSH keys (SshKeys).
+
+    Returns:
+      A DAG node that executes the model test.
+    """
+    return gpu.ssh_host.override(
+        task_id="run_model",
+        execution_timeout=datetime.timedelta(
+            minutes=self.task_test_config.time_out_in_min
+        ),
+        owner=self.task_test_config.task_owner,
+    )(
+        resource,
+        self.task_test_config.test_script,
+        ssh_keys,
+    )
+
+  def post_process(self) -> DAGNode:
+    """Process metrics and metadata, and insert them into BigQuery tables.
+
+    Returns:
+      A DAG node that executes the post process.
+    """
+    with TaskGroup(group_id="post_process") as group:
+      process_id = metric.generate_process_id.override(retries=0)()
+      metric.process_metrics.override(retries=0)(
+          process_id,
+          self.task_test_config,
+          self.task_metric_config,
+          self.task_gcp_config,
+      )
+      return group
+
+  def clean_up(self, resource: airflow.XComArg, project_id: str, zone: str) -> DAGNode:
+    """Clean up GPU resources created by `provision`.
+
+    Args:
+      resource: an XCom value for the qualified instance name.
+      project_id: project of the instance.
+      zone: zone of the instance.
+    Returns:
+      A DAG node that deletes the queued resource and its owned nodes.
+
+    Raises:
+      AirflowTaskTimeout: An error occurs when execution_timeout is breached.
+    """
+    return gpu.delete_resource.override(task_id="release_resource")(
```

Comment:
Similar comment. We should name it as `clean_up` task group, and then has a task delete_resource_request, and sensor called wait_for_resource_deletion, instead of wrapping them together: https://screenshot.googleplex.com/8uMfXiEXzoFqv62

Reply:
Thanks for the point. I have changed to task_sensor instead of the `wait_for_extended_operation` and it works well. WILL has a similar feedback regarding the resource_creation. I will make the update for that in a separate PR.